eloquentjs/4_2_reverse_array.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Performs non-mutating array reversal operation
+ * @param {*} input 
+ * @returns 
+ */
+function reverseArray(input) {
+    if (!Array.isArray(input)) {
+        throw "input must be array";
+    }
+
+    output = [];
+    for (let i = input.length-1; i >= 0; i--) {
+        output.push(input[i]);
+    }
+    return output;
+}
+
+/**
+ * Performs mutating array reversal operation
+ * @param {*} input 
+ * @returns 
+ */
+function reverseArrayInPlace(input) {
+    if (!Array.isArray(input)) {
+        throw "input must be array";
+    }
+
+    let maxIndex = input.length-1;
+    let swap;
+    for (let i = 0; i < input.length/2; i++) {
+        swap = input[i];
+        input[i] = input[maxIndex-i]
+        input[maxIndex-i]=swap;
+    }
+}
+
+let testArray = [4,5,6,7,8];
```

Comment:
For a function like reverse in place you should test with arrays that are even, odd, 0, 1, 2, and 3 elements to make sure you're catching all of the possible issues with indexes meeting up you can run into. I believe your logic is correct, but I'd still like to see the test cases.